newroutes/authenticatedSites/__tests__/Homepage.spec.js

```diff
@@ -0,0 +1,169 @@
+    read: jest.fn(),
+    update: jest.fn(),
+  }
+
+  const router = new HomepageRouter({
+    homepagePageService: mockHomepagePageService,
+  })
+
+  const app = express()
+  app.use(express.json({ limit: "7mb" }))
+  app.use(express.urlencoded({ extended: false }))
+
+  // We can use read route handler here because we don't need to lock the repo
+  app.get(
+    "/:siteName/homepage",
+    attachReadRouteHandlerWrapper(router.readHomepage)
+  )
+  app.post(
+    "/:siteName/homepage",
+    attachReadRouteHandlerWrapper(router.updateHomepage)
+  )
+  app.use(errorHandler)
+
+  const siteName = "test-site"
+  const accessToken = undefined // Can't set request fields - will always be undefined
+  const fileName = "test-file"
+  const mockSha = "12345"
+  const mockContent = "mock-content"
+
+  const reqDetails = { siteName, accessToken }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("readHomepage", () => {
+    const expectedResponse = {
+      sha: mockSha,
+      content: mockContent,
+    }
+
+    mockHomepagePageService.read.mockResolvedValue(expectedResponse)
+
+    it("retrieves homepage details", async () => {
+      const resp = await request(app).get(`/${siteName}/homepage`).expect(200)
+
+      expect(resp.body).toStrictEqual(expectedResponse)
+      expect(mockHomepagePageService.read).toHaveBeenCalledWith(reqDetails)
+    })
+  })
+
+  describe("updateHomepage", () => {
+    const updatePageDetails = {
+      content: {
+        frontMatter: {
+          layout: "homepage",
+          title: "title",
+          description: "desc",
+          permalink: "permalink",
+          notification: "notification",
+          image: "image",
+          sections: [
+            {
+              hero: {
+                title: "hero-title",
+              },
+            },
+            {
+              infobar: {
+                title: "Infobar title",
+                subtitle: "Subtitle",
+              },
+            },
+          ],
+        },
+        pageBody: mockContent,
+      },
+      sha: mockSha,
+    }
+
+    it("rejects requests with invalid body", async () => {
+      await request(app).post(`/${siteName}/homepage`).send({}).expect(400)
+    })
+
+    it("rejects requests with incomplete body", async () => {
+      // Missing layout
+      const incompleteDetails = {
+        content: {
+          frontMatter: {
+            title: "title",
+            description: "desc",
+            permalink: "permalink",
+            notification: "notification",
+            image: "image",
+            sections: [
+              {
+                hero: {
+                  title: "hero-title",
+                },
+              },
+              {
+                infobar: {
+                  title: "Infobar title",
+                  subtitle: "Subtitle",
+                },
+              },
+            ],
+          },
+          pageBody: mockContent,
+        },
+        sha: mockSha,
+      }
```

Comment:
nit: we can use `_.omit` here for clarity

Reply:
Good catch, added in https://github.com/isomerpages/isomercms-backend/pull/331/commits/39c91db6702135cc1f48bae11ed7ce1601e20639